src/Symfony/Component/Console/Tester/CommandTester.php

```diff
@@ -125,7 +125,7 @@ public function getOutput()
     /**
      * Gets the status code returned by the last execution of the application.
      *
-     * @return int The status code
+     * @return int|null The status code
```

Comment:
When can we have null?

Reply:
When you call it before the command execution ends: https://github.com/symfony/symfony/blob/813d220aa1eda4b45d545537ab12cd43622aa506/src/Symfony/Component/Process/Process.php#L728-L738

I discovered it exactly because I wanted to test that I got a `CommandTester` without `execute` being called.

Reply:
What about setting the default value to 0 instead line 32?

Reply:
`0` is "successfully completed" in terms of exit codes, IMHO is not the way to go, and inconsistent with how the commands & processes behave outside tests.

Reply:
Calling ˋgetExitCode()` before ˋexecute()` is a developer mistake, same for other getters.
I would make it throw a ˋLogicException` instead, that’d be consistent with `getDisplay()`.

Reply:
Nice suggestion! I've implemented like that, and added some tests.